lonboard/experimental/traits.py

```diff
@@ -101,3 +102,71 @@ def validate(
 
         self.error(obj, value)
         assert False
+
+
+class NormalAccessor(Union[TraitType, FixedErrorTraitType]):
+    """
+        A representation of a deck.gl normal accessor
+        
+        Acceptable inputs:
+        - A numpy ndarray with two dimensions: The size of the second dimension must be 3 i.e. `(N,3)`
+        - a pyarrow `FixedSizeListArray` or `ChunkedArray` containing `FixedSizeListArray`s 
+        where the size of the inner fixed size list 3.
+    """
+    default_value = [0,0,1]
+    info_text = (
+        "List representing normal of each object, in [nx, ny, nz]. or numpy ndarray or "
+        "pyarrow FixedSizeList representing the normal of each object, in [nx, ny, nz]"
+    )
+
+    def __init__(
+        self: TraitType,
+        *args,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.tag(sync=True, **NORMAL_SERIALIZATION)
+        
+    def validate(
+        self, obj, value
+    ) -> Union[Tuple[int, ...], List[int], pa.ChunkedArray, pa.FixedSizeListArray]:
+        
+        
+        # TODO Redundant logic?
+        list_size = value.shape[1]
+        if list_size != 3:
+            self.error(
+                obj,
+                value,
+                info="Point array must have 3 as its second dimension. I.e. (N,3)",
+            )
+        
+        if isinstance(value, np.ndarray):
+            # values must be contiguous (the same length for all values)
+            if value.ndim != 2 or value.shape[1] != 3:
+                self.error(obj, value, info="Normal array must be 2D with shape (N,3)")
+                
+            if not np.any(value == 1):
+                self.error(obj, value, info="One of the normal vector elements must be 1.")
+                
+            return pa.FixedSizeListArray.from_arrays(value.flatten("C"), list_size)
+                
+        if isinstance(value, (pa.ChunkedArray, pa.Array)):
+            if not pa.types.is_fixed_size_list(value.type):
+                self.error(
+                    obj, value, info="Point pyarrow array must be a FixedSizeList."
+                )
+                
+            if value.type.list_size != 3:
+                self.error(
+                    obj,
+                    value,
+                    info=(
+                        "Normal pyarrow array must have a FixedSizeList inner size of 3."
+                    ),
+                )
+
+            return pa.FixedSizeListArray.from_arrays(value.flatten("C"), list_size)
```

Comment:
Same thing with pyarrow arrays, we need to validate that it's of a floating point data type, and then cast it to float32. It may be easier in the numpy section to overwrite the `value` variable with a pyarrow array, which will then activate this `if` block, and you only have to cast data types once.